beacon-chain/forkchoice/protoarray/helpers_test.go

```diff
@@ -0,0 +1,325 @@
+		votes = append(votes, Vote{indexToHash(0), indexToHash(lastIndex), 0})
+		oldBalances = append(oldBalances, balance)
+		newBalances = append(newBalances, balance)
+	}
+
+	delta, _, err := computeDeltas(context.Background(), indices, votes, oldBalances, newBalances)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(delta) != int(validatorCount) {
+		t.Error("Incorrect length")
+	}
+
+	for i, d := range delta {
+		if i == 0 {
+			if d != -int(balance*validatorCount) {
+				t.Error("first root should have negative delta")
+			}
+		} else if i == int(lastIndex) {
+			if d != int(balance*validatorCount) {
+				t.Error("last root should have positive delta")
+			}
+		} else {
+			if d != 0 {
+				t.Error("Delta should be zero")
+			}
+		}
+	}
+
+	for _, vote := range votes {
+		if vote.currentRoot != vote.nextRoot {
+			t.Errorf("The vote should have changed")
+		}
+	}
+}
+
+func TestComputeDelta_MoveOutOfTree(t *testing.T) {
+	balance := uint64(32)
+	indices := make(map[[32]byte]uint64)
+	votes := make([]Vote, 0)
+	oldBalances := []uint64{balance, balance}
+	newBalances := []uint64{balance, balance}
+
+	indices[indexToHash(1)] = 0
+
+	votes = append(votes, Vote{indexToHash(1), params.BeaconConfig().ZeroHash, 0})
+	votes = append(votes, Vote{indexToHash(1), [32]byte{'A'}, 0})
+
+	delta, _, err := computeDeltas(context.Background(), indices, votes, oldBalances, newBalances)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(delta) != 1 {
+		t.Error("Incorrect length")
+	}
+
+	if delta[0] != 0-2*int(balance) {
+		t.Error("Incorrect delta")
+	}
+
+	for _, vote := range votes {
+		if vote.currentRoot != vote.nextRoot {
+			t.Errorf("The vote should have changed")
+		}
+	}
+}
+
+func TestComputeDelta_ChangingBalances(t *testing.T) {
+	oldBalance := uint64(32)
+	newBalance := oldBalance * 2
+	validatorCount := uint64(16)
+	indices := make(map[[32]byte]uint64)
+	votes := make([]Vote, 0)
+	oldBalances := make([]uint64, 0)
+	newBalances := make([]uint64, 0)
+
+	indices[indexToHash(1)] = 0
+
+	for i := uint64(0); i < validatorCount; i++ {
+		indices[indexToHash(i)] = i
+		votes = append(votes, Vote{indexToHash(0), indexToHash(1), 0})
+		oldBalances = append(oldBalances, oldBalance)
+		newBalances = append(newBalances, newBalance)
+	}
+
+	delta, _, err := computeDeltas(context.Background(), indices, votes, oldBalances, newBalances)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if len(delta) != 16 {
+		t.Error("Incorrect length")
+	}
+	for i, d := range delta {
+		if i == 0 {
+			if d != -int(oldBalance*validatorCount) {
+				t.Error("first root should have negative delta")
+			}
+		} else if i == 1 {
+			if d != int(newBalance*validatorCount) {
+				t.Error("last root should have positive delta")
+			}
+		} else {
+			if d != 0 {
+				t.Error("Delta should be zero")
+			}
+		}
+	}
+
+	for _, vote := range votes {
+		if vote.currentRoot != vote.nextRoot {
+			t.Errorf("The vote should have changed")
+		}
+	}
+}
+
+func TestComputeDelta_ValidatorAppear(t *testing.T) {
```

Comment:
What does `ValidatorAppear` in the test name mean?

Reply:
It means an incoming validator submitted a new vote that wasn't part of vote tracker